### PR TITLE
docs: add recording page

### PR DESCRIPTION
**Pull request by takeshi-iwanari**
Signed-off-by: takeshi.iwanari <takeshi.iwanari@tier4.jp>

Some contents in tutorials/recording.md are moved to recording page. To avoid conflict, I keep tutorials/recording.md at the moment, so I added tutorials/recording_modified.md where some links are fixed and some contents are removed. We can overwrite  tutorials/recording_modified.md to  tutorials/recording.md later.

### CODE REVIEW

**Review by hsgwa — CHANGES_REQUESTED** [review]
Just a small modification.

File: docs/recording/build_check.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      ros-rclcpp and the headers of CARET/rclcpp, resulting in compile errors.
+      Therefore, test codes have to be excluded from building.
+<prettier-ignore-end>
+
+## Check whether CARET/rclcpp is applied to each package
+
+You can check whether a target application is successfully built with CARET/rclcpp using `ros2 caret check_caret_rclcpp` command.
+
+```sh
+ros2 caret check_caret_rclcpp --workspace <path-to-project>
```
[the remark this answers is not in the document]
Comment (takeshi-iwanari) [reply docs/recording/build_check.md:41]: Thank you for the review.
Fixed in https://github.com/tier4/CARET_doc/pull/84/commits/64036f93b3e2781c72f830ad5a58ce564f3b6e90.

File: docs/recording/build_check.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# Build with CARET
+
+## Build a target application with CARET
+
+A target application should be built with CARET/rclcpp to record trace data. If you have already built the application without CARET/rclcpp, you have to build the application with CARET/rclcpp again.
+
+For building the application with CARET/rclcpp, CARET's `local_setup.bash` should be applied along with ROS 2's `setup.bash` as shown below. Also, `-DBUILD_TESTING=OFF` should be given to build option.
+
+```sh
+cd <path-to-project>
```
[the remark this answers is not in the document]
Comment (takeshi-iwanari) [reply docs/recording/build_check.md]: Thank you for the review.
Fixed in https://github.com/tier4/CARET_doc/pull/84/commits/64036f93b3e2781c72f830ad5a58ce564f3b6e90.

File: docs/recording/recording.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+   source ./caret_topic_filter.bash
+
+   ros2 launch caret_demos end_to_end_sample_with_lttng_session.launch.py
+   ```
+
+## Advanced: Useful settings for launch file
+
+- The following script shows advanced settings for a launch file
+- `caret_session` option is used to set a session name. If not assigned, datetime (YYYYMMDD-HHMMSS) is used
+- `caret_light` option is used to add another event filter. Only events specified in `caret_event` variable are recorded. In this exsample, high level events, such as ros2:rclcpp layer, are recorded if `caret_light` is set to "true". This setting will be helpful to record a huge application.
```
Comment (hsgwa) [inline docs/recording/recording.md:138]: typo

```suggestion
- `caret_light` option is used to add another event filter. Only events specified in `caret_event` variable are recorded. In this example, high level events, such as ros2:rclcpp layer, are recorded if `caret_light` is set to "true". This setting will be helpful to record a huge application.
```

**Review by hsgwa — APPROVED** [review]
LGTM!

File: docs/recording/recording.md (hunk 3 above)
Comment (takeshi-iwanari) [reply docs/recording/recording.md]: @hsgwa 
Thank you for correcting it. mm

**Pull request merged**